src/main/java/uk/gov/ons/ctp/response/action/service/CaseNotificationService.java

```
@@ -18,108 +17,134 @@
 
-  @Autowired private CollectionExerciseClientService collectionSvcClientServiceImpl;
+  private CollectionExerciseClientService collectionSvcClientServiceImpl;
+
+  public CaseNotificationService(
+      ActionCaseRepository actionCaseRepo,
+      ActionPlanRepository actionPlanRepo,
+      ActionService actionService,
+      CollectionExerciseClientService collectionSvcClientServiceImpl) {
+    this.actionCaseRepo = actionCaseRepo;
+    this.actionPlanRepo = actionPlanRepo;
+    this.actionService = actionService;
+    this.collectionSvcClientServiceImpl = collectionSvcClientServiceImpl;
+  }
 
   @Transactional(
       propagation = Propagation.REQUIRED,
       readOnly = false,
       timeout = TRANSACTION_TIMEOUT)
   public void acceptNotification(final CaseNotification notification) throws CTPException {
-    final String actionPlanIdStr = notification.getActionPlanId();
-    final UUID actionPlanId = UUID.fromString(actionPlanIdStr);
-    final ActionPlan actionPlan = actionPlanRepo.findById(actionPlanId);
-    final UUID caseId = UUID.fromString(notification.getCaseId());
-    final UUID collectionExerciseId = UUID.fromString(notification.getExerciseId());
-    UUID partyId = null;
-    if (notification.getPartyId() != null) {
-      partyId = UUID.fromString(notification.getPartyId());
+
+    UUID caseId = UUID.fromString(notification.getCaseId());
+
+    switch (notification.getNotificationType()) {
+      case REPLACED:
+      case ACTIVATED:
+        ActionCase actionCase = createActionCase(notification);
+        saveActionCase(actionCase);
+        break;
+
+      case ACTIONPLAN_CHANGED:
+        updateActionCase(caseId, UUID.fromString(notification.getActionPlanId()));
+        break;
+
+      case DISABLED:
+      case DEACTIVATED:
+        actionService.cancelActions(caseId);
+        deleteActionCase(caseId);
+        break;
+
+      default:
+        log.with("notification_type", notification.getNotificationType())
```

This should probably be a `throw new IllegalArgumentException("Invalid notification type")`